urllib3/connection.py

```diff
@@ -162,6 +162,27 @@ def connect(self):
         conn = self._new_conn()
         self._prepare_conn(conn)
 
+    def request_chunked(self, method, url, body=None, headers={}):
+        """ 
+        Alternative to the common request method, which sends the 
+        body with chunked encoding and not as one block 
+        """
+        header_names = set(k.lower() for k in headers)
+        self.putrequest(method, url, skip_accept_encoding=True)
+        for header, value in headers.items():
+            self.putheader(header, value)
+        if 'transfer-encoding' not in headers:
+            self.putheader('Transfer-Encoding', 'chunked')
```

Comment:
I was referring to that code: https://github.com/python/cpython/blob/2.7/Lib/httplib.py#L1005 but I had remembered it wrong, it's the accept-encoding that get's set. Looking at the comments there, looks like we're not the only ones who are bit undecided about what to set automatically.